.github/workflows/kickstart-tests.yml

```diff
@@ -0,0 +1,127 @@
+          echo "::set-output name=allowed_user::true"
+
+      - name: Get information for pull request
+        uses: octokit/request-action@v2.x
+        id: pr_api
+        with:
+          route: GET /repos/${{ github.repository }}/pulls/${{ github.event.issue.number }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Parse test configuration
+        id: parse_test_config
+        run: |
+          TESTTYPE=$(echo "${{ github.event.comment.body }}" | head -n1 | awk '{print $2}')
+          echo "::set-output name=testtype::${TESTTYPE}"
+
+    outputs:
+      allowed_user: ${{ steps.check_user_perm.outputs.allowed_user }}
+      base_ref: ${{ fromJson(steps.pr_api.outputs.data).base.ref }}
+      sha: ${{ fromJson(steps.pr_api.outputs.data).head.sha }}
+      testtype: ${{ steps.parse_test_config.outputs.testtype }}
+
+  run:
+    needs: pr-info
+    # only do this for master for now; once we have RHEL 8/9 boot.iso builds working, also support these
+    if: needs.pr-info.outputs.base_ref == 'master' && needs.pr-info.outputs.allowed_user == 'true' && needs.pr-info.outputs.testtype != ''
+    runs-on: [self-hosted, kstest]
+    timeout-minutes: 300
+    steps:
+      # self-hosted runners don't do this automatically; also useful to keep stuff around for debugging
+      # need to run sudo as the launch script and the container create root/other user owned files
+      - name: Clean up previous run
+        run: |
+          sudo podman ps -q --all --filter='ancestor=kstest-runner' | xargs -tr sudo podman rm -f
+          sudo podman volume rm --all || true
+          sudo rm -rf * .git
+
+      # we post statuses manually as this does not run from a pull_request event
+      # https://developer.github.com/v3/repos/statuses/#create-a-status
+      - name: Create in-progress status
+        uses: octokit/request-action@v2.x
+        with:
+          route: 'POST /repos/${{ github.repository }}/statuses/${{ needs.pr-info.outputs.sha }}'
+          context: 'kickstart ${{ needs.pr-info.outputs.testtype }}'
+          state: pending
+          target_url: 'https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ needs.pr-info.outputs.sha }}
+          fetch-depth: 0
+
+      - name: Rebase to current master
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git log --oneline -1 origin/master
+          git rebase origin/master
+
+      # FIXME: build a complete boot.iso with lorax/osbuild
+      - name: Build updates.img
+        run: |
+          scripts/makeupdates
+          gzip -cd updates.img | cpio -tv
+
+      - name: Check out kickstart-tests
+        run: git clone --depth=1 https://github.com/rhinstaller/kickstart-tests
```

Comment:
Please use checkout action here. I don't think there is a reason to call ``git clone`` manually.

Reply:
OK. I originally didn't because I don't want it to overwrite the anaconda checkout -- but I see that the action has a `path:` option. I'm happy to use that if you prefer.

Reply:
Done.